R/bootstrap.nlsfit.R

```diff
@@ -211,44 +251,24 @@ print.bootstrapfit <- function(x, digits=2, ...) {
   summary.bootstrapfit(object=x, digits=digits, ...)
 }
 
-plot.bootstrapfit <- function(x, ..., xlim, ylim, rep=FALSE, col.line="black", col.band="gray", lty=c(1), lwd=c(1), xlab="x", ylab="y") {
+plot.bootstrapfit <- function(x, ..., xlim, ylim, rep=FALSE, col.line="black", col.band="gray", lty=c(1), lwd=c(1), xlab="x", ylab="y", samples=1000) {
   rx <- range(x$x)
-  X <- seq(rx[1], rx[2], (rx[2]-rx[1])/1000)
+  X <- seq(rx[1], rx[2], (rx[2]-rx[1])/samples)
   npar <- length(x$par.guess)
-  Y <- numeric()
-  my.xlim <- numeric()
-  my.ylim <- numeric()
-  if(!missing(xlim)) {
-    my.xlim <- xlim
-  }
-  if(!missing(ylim)) {
-    my.ylim <- ylim
+
+  ## use the xylimits computation of plotwitherror
+  if(x$errormodel == "yerrors") {
+    mylims <- plotwitherror(x=x$x, y=x$y, dy=x$dy, rep=rep, xlim=xlim, ylim=ylim, ...)
   }
-  if(!rep) {
-    mylims <- c()
-    ## use the xylimits computation of plotwitherror
-    if(missing(xlim) || missing(ylim)) {
-      if(x$errormodel == "yerrors") {
-        mylims <- plotwitherror(x=x$x, y=x$y, dy=x$dy, ...)
-      }
-      else {
-        mylims <- plotwitherror(x=x$x, y=x$y, dy=x$dy, dx=x$dx, ...)
-      }
-    }
-    if(missing(xlim)) {
-      my.xlim <- mylims$xlim
-    }
-    if(missing(ylim)) {
-      my.ylim <- mylims$ylim
-    }
-    
-    ## generate empty plot
-    
-    plot(NA, xlim=my.xlim, ylim=my.ylim, xlab=xlab, ylab=ylab, ...)
+  else {
+    mylims <- plotwitherror(x=x$x, y=x$y, dy=x$dy, dx=x$dx, rep=rep, xlim=xlim, ylim=ylim, ...)
```

Comment:
I did the `plotwitherror` at the end of everything on purpose, such that the data points are on top of everything else. It seems this is changed now?

Reply:
Yes, I changed this for the following reasons:
-In the way it has been, there have been always two plots created when using a script and producing a pdf. The first one contained only the data, the second one the fit and the data.
-If you have a lot of data and plot it on top of the fit, you don't see the fit any more. But if you plot the data first and then the fitline you still see everything because the errorband is transparent.
-The way it is now is consistent with plot.effectivemass where the fit is plotted on top as well.
-With this function one wants to plot the results of the fit, not the data, so this is where the focus should lie.